File: aspnet-core/src/SavageRuler.Core/Rules/Powers/Power.cs
using Abp.Auditing;
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SavageRuler.Rules.Powers
{
    [Audited]
    public class Power : SavageRulerEntity
    {
        [Required(AllowEmptyStrings = true), MaxLength(64)]
        public virtual string Name { get; set; }
        public virtual string Book { get; set; }
        public virtual string Points { get; set; }
        public virtual string Duration { get; set; }
        public virtual string Distance { get; set; }
        public virtual string Rank { get; set; }
        public virtual string Trappings { get; set; }
        public virtual string Text { get; set; }
    }
}

File: ng/src/app/api/entities/powers/descriptors/power.ts
import {IEntity} from '../../../../types/api/ientity';

export interface Power extends IEntity<number> {
  name: string;
  book: string;
  points: string;
  duration: string;
  distance: string;
  rank: string;
  trappings: string;
  text: string;
}
